aws/awserr/types.go

```diff
@@ -2,6 +2,16 @@ package awserr
 
 import "fmt"
 
+const (
+	// CodeSerialization is the serialization error code that is received
+	// during protocol unmarshaling.
+	CodeSerialization = "SerializationError"
```

Comment:
`ErrCode` prefix should be used for these error codes is what were used for the service clients error codes.

We probably should also define these in the `request` package. That is where the other request errors are enumerated.